l10n-dev/src/ast/unescapeString.ts

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * Replaces escaped sequences in the string, including:
+ *
+ *  - Standard escapes (\r, \n, \t, and standard chars)
+ *  - Unicode escapes (\xFF, \uFFFF, \u{10FFFF})
+ *
+ * Its behavior is undefined if the `text` is not a valid ECMAScript string.
+ *
+ * @see https://exploringjs.com/es6/ch_unicode.html
+ */
+export const unescapeString = (text: string): string => {
```

Comment:
This seemed less sketchy that eval 😛

Reply:
this worries me because I had many iterations of trying to get this right and settled with eval as the nuclear approach... be we can give it a try!

Reply:
The test cases seemed to work, and I verified that it produced identical output when run against `js-debug`